### DISCUSSION TITLE
Indexing for labels problem

### DISCUSSION THREAD

**Discussion by alihsan-tsdln** [JanusGraph/janusgraph#4627] [Q&A] [answered]
I am implemeting Spark Datasource for JanusGraph. I should fetch data from JanusGraph with Spark DataSource reader. My issue is fetching data from JanusGraph without indexing. It makes me so slow. I should take vertex data by label but I cannot implement indexing on it. What do you recommend for it? Normally, I put property key as "vertex_type" and add same label name on it but for general use datasource I cannot expect same thing from general user. What should I do?

**Comment by sirohiwebdev** [comment, marked as answer]
Indexing is not supported on Vertex labels in JanusGraph, there is already an issue created for the same https://github.com/JanusGraph/janusgraph/issues/283
You need to go by property key only which represents the label.
